### PR TITLE
Fix makeArray for types with disabled this(this)

### PR DESCRIPTION
**Pull request by ZombineDev**

### CODE REVIEW

**Comment by JakobOvrum** [comment]
Nice, LGTM

**Comment by ZombineDev** [comment]
@JakobOvrum fixed & rebased. Yeah, I use editorconfig, but I did this change through the GitHub interface which unfortunately doesn't highlight deviations from editorconfig.

**Comment by ZombineDev** [comment]
@MartinNowak any chance to cherrypick this into stable, before the release? I need to use LDC/GDC for performance I would prefer not to have to wait 4-12 months for them to catch up.

File: std/experimental/allocator/package.d [hunk 1]
```diff
@@ -573,11 +573,22 @@ unittest
 
 private T[] uninitializedFillDefault(T)(T[] array) nothrow
 {
-    static immutable __gshared T t;
+    T t = T.init;
     fillWithMemcpy(array, t);
     return array;
 }
 
+pure nothrow @nogc
```
Comment (JackStouffer) [inline std/experimental/allocator/package.d]: style nitpick: these should be on the same line as the `unittest`

**Comment by JackStouffer** [comment]
LGTM

File: std/experimental/allocator/package.d (hunk 1 above)
Comment (ZombineDev) [inline std/experimental/allocator/package.d]: The attribute style is totally inconsistent throughout the module, I didn't know which to pick, so I did this, which will be consistent with a future PR of mine which adds missing attributes to a lot of the stuff.

Comment (ZombineDev) [inline std/experimental/allocator/package.d]: And by the way there should be a rule about this in the [Phobos styleguide](https://dlang.org/dstyle.html).

Comment (JackStouffer) [inline std/experimental/allocator/package.d]: Ok, this is fine as is then

Comment (ZombineDev) [inline std/experimental/allocator/package.d]: See also [PR 3957](https://github.com/D-Programming-Language/phobos/pull/3957).

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Comment by ZombineDev** [comment]
Thanks @JakobOvrum and @JackStouffer for the review!

**Pull request merged**